### ISSUE TITLE
[Feature Request] UI: Dashboard lifecycle hooks

### ISSUE THREAD

**Issue by devaskim** [thingsboard/thingsboard#7465]
**Describe the solution you'd like**

The main idea is to provide similar (but I think not the same) hooks/callbacks as it is already done for custom widgets.

**Is your feature request related to a problem? Please describe.**

While developing custom widgets and dashboards I face with various scenarios where dashboard hooks would simplify implementation. Here is one of them:

We have dashboard (called *Plant*) to show various information of *single* plant. Customer user could open it directly if he/she owns only *one* plant. If customer has several plants, he/she will open *Plant* dashboard from another dashboard that lists all plants.

So on this stage I need 2 different alias for *Plant* dashboard to resolve plant entity. In case of *single* plant it is, e.g., *Asset type* alias, but for multiple plants' case it is *Entity from dashboard state* alias. Thus I made 2 states with the same widget set for *Plant* dashboard where 2 aliases mentioned above are used.

But *Plant* dashboard has another states where I wish to use *single* alias to resolve map entity instead of duplicating widgets. So I needed way to "merge" 2 aliases into 1. I found way that I described in *Alternatives* section of this issue.

**Describe alternatives you've considered**
I added custom widget as a header to all states of *Plant* dashboard. 
![header_widget](https://user-images.githubusercontent.com/86909129/197142484-54a83a27-86f7-4781-946c-3698cab492dc.jpg)

There, in `onInit` widget hook/callback, I manually update dashboard state if it is empty. In this way I initialize *Entity from dashboard state* alias that is used in all other dashboard states.
```
    let stateParams = self.ctx.stateController ? self.ctx.stateController.getStateParams() : null;
    if (stateParams && !('entityId' in stateParams)) {
        stateParams.entityId = {
            id: self.ctx.datasources[0].entityId,
            entityType: self.ctx.datasources[0].entityType,
            entityName: self.ctx.datasources[0].entityName,
            entityLabel: self.ctx.datasources[0].entityLabel
        };
        self.ctx.stateController.updateState(null, stateParams);
        self.ctx.aliasController.dashboardStateChanged();
    }
```
 In other words, when user who owns only single plant will directly open *Plant* dashboard. And both *Asset type* alias and *Entity from dashboard state* will be initialized with the same entity data.

**Conclusion**

I think such logic could be moved to dashboard lifecycle  hooks.
